src/main/java/tech/stackable/gis/hbase/coprocessor/TopXQueue.java

```diff
@@ -0,0 +1,22 @@
+package tech.stackable.gis.hbase.coprocessor;
+
+import java.util.PriorityQueue;
+
+public class TopXQueue extends PriorityQueue<Item> {
+    private static final long serialVersionUID = 5134028249611535803L;
+    private final int count;
+
+    public TopXQueue(int count) {
+        super(count);
+        this.count = count;
+    }
+
+    @Override
+    public boolean add(Item item) {
+        super.add(item);
+        if (size() > count) {
+            poll(); // remove smallest i.e. oldest value
```

Comment:
This method is not thread safe. I don't know if it's relevant.

Reply:
I'll add to this to the optimization issue.